lib/export/formatters/dataElementFormatterCimpl6.js

```diff
@@ -0,0 +1,688 @@
+            continue;
+          }
+
+          if (['code', 'Coding'].includes(p.name)) {
+            continue;
+          }
+
+          combinedPaths.push(p.name);
+        }
+      }
+
+      let combinedPath = (combinedPaths.length > 0) ? combinedPaths.join('.') : 'root';
+
+      if (!out[combinedPath]) out[combinedPath] = [];
+      out[combinedPath].push(constraint);
+
+      return out;
+    }, {});
+
+    if (Object.keys(constraintsByPaths).length && Object.keys(constraintsByPaths)[0] == 'CodeableConcept') {
+      constraintsByPaths = {'root':constraintsByPaths['CodeableConcept']};
+    }
+    return constraintsByPaths;
+  }
+
+  formatFields(de) {
+    const formattedFields = [];
+    const constrainedFields = [];
+    for (const f of de.fields) {
+      if (f.inheritance != 'inherited') {
+        const constraintsByPaths = this.arrangeValuesByConstraintPaths(f, de);
+        if (Object.keys(constraintsByPaths).length > 0) {
+          for (const path in constraintsByPaths) {
+            const fClone = f.clone();
+            fClone.constraints = constraintsByPaths[path];
+            let inserted_value = this.formatValue(fClone, de, true, path);
+            if (f.inheritance == 'overridden') {
+              for (var key in this.substituteHash) {
+                if (inserted_value.includes(key) && !inserted_value.includes('substitute')) {
+                  inserted_value = inserted_value.replace(key, this.substituteHash[key])
+                }
+              }
+              let constrainedTerm = `${rightPad('', this.maxPlacement)}${inserted_value.trim()}`;
+              constrainedFields.push(`${rightPad('', this.maxPlacement)}${inserted_value.trim()}`);
+            }
+            else {
+              if (fClone.constraints.length > 0) {
+                formattedFields.push(`${rightPad('Property:', this.maxPlacement - 7)}${fClone.identifier.name} ${fClone.card.toString()}`);
+                let i = 0;
+                for (i = 0; i < fClone.constraints.length; i++) {
+                  if (fClone.constraints[i].valueSet !== undefined) {
+                    let vs = fClone.constraints[i].valueSet;
+                    let vs_object = this.specs._valueSets.findByURL(vs);
+                    if (vs_object !== undefined) {
+                      vs = vs_object.identifier.name;
+                    }
+                    constrainedFields.push(`${rightPad('', this.maxPlacement)}${fClone.identifier.name} from ${vs} (${fClone.constraints[i].bindingStrength.toLowerCase()})`);
+                  }
+                }
+              }
+              else {
+                formattedFields.push(`${rightPad('Property:', this.maxPlacement - 7)}${inserted_value.trim()} ${fClone.card.toString()}`);
+              }
+            }
+          }
+        } else {
+          let inserted_value = this.formatValue(f, de, true);
+          if (f.inheritance == 'overridden') {
+            for (var key in this.substituteHash) {
```

Comment:
Same comments as above re: `this.substituteHash` and preferring `let` over `var`.